packages/modal/src/ModalContainer.ts

```diff
@@ -7,12 +7,21 @@
 
 import PropTypes from 'prop-types';
 
-import { useModal } from './useModal';
+import { useModal, IUseModalProps, IUseModalReturnValue } from './useModal';
 
-export function ModalContainer({ children, render = children, ...options }) {
-  return render(useModal(options));
+export interface IModalContainerProps extends IUseModalProps {
+  render?: (options: IUseModalReturnValue) => any;
+  children?: (options: IUseModalReturnValue) => any;
```

Comment:
In the past TypeScript migrations, render prop functions had a [return](https://github.com/zendeskgarden/react-containers/blob/master/packages/focusjail/src/FocusJailContainer.ts#L13) [type](https://github.com/zendeskgarden/react-containers/blob/master/packages/tabs/src/TabsContainer.ts#L14) of `React.ReactElement`. However, here I am annotating the functions to return an `any` type. I did this because a consumer of the container can return any value (not just React elements):

Example 1
```jsx
<ContainerComp>
  {(propGetters) => {
    return isFoo ? <div>foo</div> : null  // can return null
  }}
</ContainerComp>
```

Example 2
```jsx
<ContainerComp>
  {(propGetters) => {
    return isFoo && <div>foo</div> // can return a boolean
  }}
</ContainerComp>
```

Here is an [example use case](https://github.com/zendeskgarden/react-containers/blob/master/packages/modal/stories.js#L100) similar to Example 2 in the current `master` branch. Let me know what ya'll think. I am considering going back and modifying the render prop function return types to be of `any` in a separate pull request.

Reply:
@austin94 and I have determined that the proper type annotation for render and children as a function is for it to return a type `React.ReactNode` and container function component return value as `React.ReactElement`.

That would mean the code would look like this:

```jsx
export interface IModalContainerProps extends IUseModalProps {
  render?: (options: IUseModalReturnValue) => React.ReactNode;
  children?: (options: IUseModalReturnValue) => React.ReactNode;
}

export const ModalContainer: React.FunctionComponent<IModalContainerProps> = ({
  children,
  render = children,
  ...options
}) => {
  return render!(useModal(options)) as React.ReactElement;
};
```

⚠️ Currently other parts of `master` that's annotating render prop functions to return `any` or `React.ReactElement`. These render prop functions will have to be updated to have a return type of `React.ReactNode`. [Here](https://github.com/zendeskgarden/react-containers/blob/master/packages/selection/src/SelectionContainer.ts#L14) [are](https://github.com/zendeskgarden/react-containers/blob/master/packages/accordion/src/AccordionContainer.ts#L13) [some](https://github.com/zendeskgarden/react-containers/blob/master/packages/buttongroup/src/ButtonGroupContainer.ts#L13) [examples](https://github.com/zendeskgarden/react-containers/blob/master/packages/tabs/src/TabsContainer.ts#L13) of places that need to be updated.